diffusion/datasets/image_caption.py

```diff
@@ -154,31 +163,26 @@ def __getitem__(self, index):
                 caption = random.sample(caption, k=1)[0]
             out['drop_caption_mask'] = 1.0
 
-        max_length = None if self.sdxl else self.tokenizer.model_max_length  # type: ignore
-        tokenizer_out = self.tokenizer(caption,
-                                       padding='max_length',
-                                       max_length=max_length,
-                                       truncation=True,
-                                       return_tensors='pt')
-        if self.sdxl:
-            tokenized_caption = [tokenized_cap.squeeze() for tokenized_cap in tokenizer_out.input_ids]
-            tokenized_caption = torch.stack(tokenized_caption)
-            # Take union over both tokenizers padding masks
-            attention_masks = tokenizer_out.attention_mask
-            attention_mask = torch.logical_or(attention_masks[0], attention_masks[1]).to(attention_masks[0].dtype)
-        else:
-            tokenized_caption = tokenizer_out.input_ids.squeeze()
-            attention_mask = tokenizer_out.attention_mask
-        out['captions'] = tokenized_caption
-        out['attention_mask'] = attention_mask
+        out['captions'] = []
+        out['attention_mask'] = []
+        for tokenizer in self.tokenizers:
+            tokenizer_out = tokenizer(caption,
+                                      padding='max_length',
+                                      max_length=tokenizer.model_max_length,
+                                      truncation=True,
+                                      return_tensors='pt')
+
+            out['captions'].append(tokenizer_out.input_ids.squeeze())
+            out['attention_mask'].append(tokenizer_out.attention_mask)
+        # TODO: do I need to take the union of the masks?
```

Comment:
yes i think so, because this gets passed into the unet as `encoder_attention_mask` which is expected to be (batch, sequence_length)